crates/bifrost/src/providers/replicated_loglet/network.rs

```diff
@@ -195,34 +221,75 @@ impl RequestPump {
     async fn get_loglet<T: TransportConnect>(
         &self,
         provider: &ReplicatedLogletProvider<T>,
+        peer_version: &PeerMetadataVersion,
         header: &CommonRequestHeader,
     ) -> Result<Arc<ReplicatedLoglet<T>>, SequencerStatus> {
-        let loglet = match provider.get_active_loglet(header.log_id, header.segment_index) {
-            Some(loglet) if loglet.params().loglet_id == header.loglet_id => loglet,
-            Some(_) => return Err(SequencerStatus::LogletIdMismatch),
-            None => {
-                let logs = self.metadata.logs();
-                let chain = logs
-                    .chain(&header.log_id)
-                    .ok_or(SequencerStatus::UnknownLogId)?;
-
-                let segment = chain
-                    .iter()
-                    .rev()
-                    .find(|segment| segment.index() == header.segment_index)
-                    .ok_or(SequencerStatus::UnknownSegmentIndex)?;
-
-                provider
-                    .get_or_create_loglet(
-                        header.log_id,
-                        header.segment_index,
-                        &segment.config.params,
+        let mut current_logs_version = provider.networking().metadata().logs_version();
+        let request_logs_version = peer_version.logs.unwrap_or(current_logs_version);
```

Comment:
No need really for current version to be mutable. Just get the latest when you need it.

Reply:
The point is i really wanna compare with the version known before I do the `get_loglet` because this can fail, then version syncs (from some where else) and then when i compare i fall into the `else`